cpp/tests/structure/mg_induced_subgraph_test.cpp

```diff
@@ -0,0 +1,311 @@
+        false),
+      cugraph::logic_error);
+#endif
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle_->get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG InducedSubgraph took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    if (induced_subgraph_usecase.check_correctness) {
+#if 0
+      d_subgraph_vertices = cugraph::test::device_gatherv(
+        *handle_,
+        raft::device_span<vertex_t const>(d_subgraph_vertices.data(), d_subgraph_vertices.size()));
+      d_subgraph_edgelist_majors = cugraph::test::device_gatherv(
+        *handle_,
+        raft::device_span<vertex_t const>(d_subgraph_edgelist_majors.data(),
+                                          d_subgraph_edgelist_majors.size()));
+      d_subgraph_edgelist_minors = cugraph::test::device_gatherv(
+        *handle_,
+        raft::device_span<vertex_t const>(d_subgraph_edgelist_minors.data(),
+                                          d_subgraph_edgelist_minors.size()));
+
+      if (d_subgraph_edgelist_weights)
+        *d_subgraph_edgelist_weights = cugraph::test::device_gatherv(
+          *handle_,
+          raft::device_span<weight_t const>(d_subgraph_edgelist_weights->data(),
+                                            d_subgraph_edgelist_weights->size()));
+
+      d_subgraph_edge_offsets = cugraph::test::device_gatherv(
+        *handle_,
+        raft::device_span<size_t const>(d_subgraph_edge_offsets.data(),
+                                        d_subgraph_edge_offsets.size()));
+
+      auto [h_offsets, h_indices, h_weights] =
+        cugraph::test::graph_to_host_csr(*handle_, mg_graph_view);
+
+      if (my_rank == 0) {
+        auto h_subgraph_vertices = cugraph::test::to_host(*handle_, d_subgraph_vertices);
+
+        auto h_cugraph_subgraph_edgelist_majors =
+          cugraph::test::to_host(*handle_, d_subgraph_edgelist_majors);
+        auto h_cugraph_subgraph_edgelist_minors =
+          cugraph::test::to_host(*handle_, d_subgraph_edgelist_minors);
+        auto h_cugraph_subgraph_edgelist_weights =
+          cugraph::test::to_host(*handle_, d_subgraph_edgelist_weights);
+        auto h_cugraph_subgraph_edge_offsets =
+          cugraph::test::to_host(*handle_, d_subgraph_edge_offsets);
+
+        induced_subgraph_validate(h_offsets,
+                                  h_indices,
+                                  h_weights,
+                                  h_subgraph_offsets,
+                                  h_subgraph_vertices,
+                                  h_cugraph_subgraph_edgelist_majors,
+                                  h_cugraph_subgraph_edgelist_minors,
+                                  h_cugraph_subgraph_edgelist_weights,
+                                  h_cugraph_subgraph_edge_offsets);
```

Comment:
So, we typically compare the SG result with the MG result in multi-GPU testing (and compare with the reference code output in SG testing). If I am not mistaken, here, we are comparing with the reference code output in MG as well. Any motivation for this?

Reply:
Too lazy to write two different comparison functions :-)

It felt like better code reuse just to validate against the reference implementation rather than write a second independent test.

Reply:
So, the initial motivation is that, we may run with larger data on MG, and comparing with the SG result will allow testing with bigger data than a sequential reference implementation. When I test something on multi-GPUs, I sometimes override the rmat_scale to the largest that fit within a single GPU to validate the MG results against the SG.

Do you want to move away from this convention (and do the same for other tests as well) or may compare with the reference just for induced subgraph?

Reply:
That is not something that I have been doing when testing MG, but I can see the value in that.  I will look at recasting this test to compare against SG.

Reply:
Latest push has refactored this.